### PR TITLE
[docker-base]: Add less into docker-base. Add strace in docker_base_dbg. Add configuration for vim

### PR DESCRIPTION
**Pull request by pavel-shirshov**

### CODE REVIEW

File: dockers/docker-base/Dockerfile.j2 [hunk 1]
```diff
@@ -25,12 +25,14 @@ RUN apt-get clean && apt-get update
 RUN apt-get -y install \
     rsyslog            \
     supervisor         \
-    vim-tiny           \
+    vim                \
```
Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: 400k vs 28M, do you have some strong requirement for a complete vim?

Comment (kcudnik) [inline dockers/docker-base/Dockerfile.j2]: We are installing N 200MB dockers anyway :)

Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: With the help of layer reuse and docker squash (https://docs.docker.com/engine/reference/commandline/build/), the total disk size of the docker images and containers in a production switches is about 570M. Currently we are still deploying images for small disk switches.

Comment (lguohan) [inline dockers/docker-base/Dockerfile.j2]: is it 2M or 28M?

Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: The largest component for vim is vim-runtime 25.5M: https://packages.debian.org/jessie/vim-runtime. The total size for vim packages is ~28M.

Comment (lguohan) [inline dockers/docker-base/Dockerfile.j2]: given the smallest disk we have to deal with is still 1.5GB, i recommend to keep vim in the debug build.

Comment (pavel-shirshov) [inline dockers/docker-base/Dockerfile.j2]: Keep vim.tiny for docker-base. But put .vimrc and make link from vim.tiny to vim.

**Pull request merged**